docs/pages/modules/module-api.mdx

```diff
@@ -483,6 +486,102 @@ The implementation for three Either types is currently provided out of the box,
 
 </APIBox>
 
+## Guides
+
+<APIBox header="Sending events">
+
+While the JavaScript -> Native communication is mostly covered by native functions, you might also want to let the JavaScript code know about some certain system events,
+for example when the clipboard content changes.
+
+To do this, in the module definition you need to provide the event names that the module can send, using the [Events](#events) definition component.
+After that you can use `sendEvent(eventName, payload)` function on the module instance to send the actual event with some payload.
+A minimal clipboard implementation that sends native events may look like this:
+
+<CodeBlocksTable>
+
+```swift
+let CLIPBOARD_CHANGED_EVENT_NAME = "onClipboardChanged"
+
+public class ClipboardModule: Module {
+  public func definition() -> ModuleDefinition {
+    Events(CLIPBOARD_CHANGED_EVENT_NAME)
+
+    OnStartObserving {
+      NotificationCenter.default.addObserver(
+        self,
+        selector: #selector(self.clipboardChangedListener),
+        name: UIPasteboard.changedNotification,
+        object: nil
+      )
+    }
+
+    OnStopObserving {
+      NotificationCenter.default.removeObserver(
+        self,
+        name: UIPasteboard.changedNotification,
+        object: nil
+      )
+    }
+  }
+
+  @objc
+  private func clipboardChangedListener() {
+    sendEvent(CLIPBOARD_CHANGED_EVENT_NAME, [
+      "contentTypes": availableContentTypes()
+    ])
+  }
+}
+```
+
+```kotlin
+const val CLIPBOARD_CHANGED_EVENT_NAME = "onClipboardChanged"
+
+class ClipboardModule : Module() {
+  override fun definition() = ModuleDefinition {
+    Events(CLIPBOARD_CHANGED_EVENT_NAME)
+
+    OnStartObserving {
+      clipboardManager?.addPrimaryClipChangedListener(listener)
+    }
+
+    OnStopObserving {
+      clipboardManager?.removePrimaryClipChangedListener(listener)
+    }
+  }
+
+  private val clipboardManager: ClipboardManager?
+    get() = appContext.reactContext?.getSystemService(Context.CLIPBOARD_SERVICE) as? ClipboardManager
+
+  private val listener = ClipboardManager.OnPrimaryClipChangedListener {
+    clipboardManager?.primaryClipDescription?.let { clip ->
+      this@ClipboardModule.sendEvent(
+        CLIPBOARD_CHANGED_EVENT_NAME,
+        bundleOf(
+          "contentTypes" to availableContentTypes(clip)
+        )
+      )
+    }
+  }
+}
+```
+
+</CodeBlocksTable>
+
+To subscribe to these events in JavaScript, you need to wrap the native module with `EventEmitter` class, like this:
+
+```ts TypeScript
```

Comment:
In the sentence above, JavaScript is being referenced. Here, in the code snippet, we are specifying TypeScript. I know the only difference between the two here is the added type but somebody not familiar with TypeScript might not get the difference and could lead to confusion.

I do think that 
- we can either reference JavaScript/TypeScript and and then also add a JavaScript example for one to copy-paste it.
- or stick to JavaScript example (and remove the type from this example) because in this page, above, we are also referencing [another JavaScript example](https://docs.expo.dev/modules/module-api/#function)

Reply:
We used "JavaScript" to refer to the JavaScript runtime (and the final code is always JavaScript), but I agree that using it interchangeably might be confusing.
I'm more in favor of using `JavaScript/TypeScript` in text and only TypeScript in the code blocks, sometimes we may want to refer to types that we're exporting and I guess it would have some value for readers 🤔

Reply:
I agree with you on referencing types that we export in different scenarios. Mentioning the `JavaScript/TypeScript` makes more sense. I think, in general, if we don't talk or provide reference types, there will be more gotchas/blockers.

I will keep an eye if our users get stuck with TypeScript/JavaScript code conversion or if there are strong opinions so that we can find a way to provide a better experience, but for now, I think the changes you made are all good!